lib/backend.ts

```diff
@@ -13,14 +18,115 @@ interface Flow {
     startBlockNumber: number,
     flowRate: number, 
     id: number
-    flowRateUpdaetEvents: any[],
+    flowUpdatedEvents: any[],
 }
 
 interface IBackend {
-    getGates(address: string): Gate[], 
+    getGates(address: string): Promise<Gate[]>, 
     getGate(address: string, gateId: number),
-    addGate(name: string, token: string, flowRate: number),
+    addGate(signer: Signer, name: string, token: string, flowRate: number),
+    checkIn(signer: Signer, gateId: string),
     deleteGate(gateId: string),
 }
 
-export type {Gate, Flow, IBackend}
\ No newline at end of file
+export class Backend implements IBackend {
+    getStreamsQuery = `
+    query GetStreams($r: String) {
+        streams(where: {receiver: $r}) {
+          id
+          streamedUntilUpdatedAt
+          updatedAtBlockNumber
+          updatedAtTimestamp
+          createdAtBlockNumber
+          createdAtTimestamp
+          currentFlowRate
+          deposit
+          sender {
+            id
+          }
+          flowUpdatedEvents {
+            oldFlowRate
+            flowRate
+            blockNumber
+            totalAmountStreamedUntilTimestamp
+            timestamp
+          }
+        }
+      }
+    `;
+
+    gql: Client;
+    pr: any;
+    constructor(graphQLClient: Client, prContract: ethers.Contract) {
+        this.gql = graphQLClient;
+        this.pr = prContract;
+    }
+
+    async getGates(receiverAddress: string): Promise<Gate[]> {
+        console.log("address", receiverAddress);
+        const response = await this.gql.query(
+            this.getStreamsQuery, 
+            {r: receiverAddress}
+        ).toPromise(); 
+        
+        const streams = response.data.streams;
+        const gates: Gate[] = [];
+
+        console.log("streams", streams);
+
+        const rawGates = await this.pr.getGates(receiverAddress);
+
+        console.log("raw gates", rawGates);
+
+        for (let index = 0; index < rawGates.length; index++) {
+            const g = rawGates[index];
+
+            gates.push({
+                name: g.name,
+                payee: receiverAddress,
+                activeFlows: this.getFlows(g, streams),
+                id: 0,
```

Comment:
We should add a comment or something indicating that this `id: 0` is a placeholder